src/components/course/data/hooks.jsx

```diff
@@ -497,16 +489,7 @@ export const useTrackSearchConversionClickHandler = ({ href, eventName }) => {
  *
  * @returns Click handler function for clicks on enrollment buttons.
  */
-export const useOptimizelyEnrollmentClickHandler = ({ href }) => {
-  const {
-    state: {
-      activeCourseRun: { key: courseKey },
-    },
-  } = useContext(CourseContext);
-  const {
-    courseEnrollmentsByStatus,
-  } = useContext(CourseEnrollmentsContext);
-
+export const useOptimizelyEnrollmentClickHandler = ({ href, courseKey, courseEnrollmentsByStatus }) => {
```

Comment:
nit: `courseKey` -> `courseRunKey` to be more explicit?

Reply:
I changed the param but kept the key in event as `courseKey`.